src/WCSimPrimaryGeneratorAction.cc

```diff
@@ -111,6 +174,7 @@ WCSimPrimaryGeneratorAction::~WCSimPrimaryGeneratorAction()
              << " = " << _counterRock/(G4double)_counterCublic << G4endl;
   }
   inputFile.close();
+  inputCosmicsFile.close();
   delete particleGun;
   delete MyGPS;   //T. Akiri: Delete the GPS variable
   delete messenger;
```

Comment:
```suggestion
  delete messenger;
  delete hFluxCosmics;
  delete hEmeanCosmics;
```
Noitced a couple of histograms that should be cleared up in the destructor

Reply:
Good spot! done, see 4fc0e54e2a93c7acb5fb2ac65c143299335982fc